libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2696,6 +2860,19 @@ int WaveTrack::GetNumClips() const
    return mClips.size();
 }
 
+int WaveTrack::GetNumClips(double t0, double t1) const
+{
+   const auto clips = SortedClipArray();
+   const auto firstIn =
+      std::find_if(clips.begin(), clips.end(), [t0](const auto& clip) {
+         return t0 < clip->GetPlayEndTime();
+      });
+   const auto firstOut = std::find_if(firstIn, clips.end(), [t1](const auto& clip) {
+      return t1 < clip->GetPlayStartTime();
+   });
```

Comment:
Maybe `std::lower_bound` instead of `find_if` for logarithmic not linear complexity in the number of clips.  After all we are encouraging the making of many little clips.  At least that's what I saw in the demo of the prototype.